inference/models/inference.py

```diff
@@ -0,0 +1,292 @@
+            self.save()
+
+    def cleanup_external_jobs(self):
+        """Delete all external jobs"""
+        if self.completed:
+            self.external_jobs.all().delete()
+
+    def set_error(self, error_msg: str) -> None:
+        """Set general error and mark job as failed"""
+        self.error_message = error_msg
+        self.status = InferenceJobStatus.FAILED
+        self.completed_at = timezone.now()
+        self.save(update_fields=["error_message", "status", "completed_at", "updated_at"])
+
+    def initiate(self) -> None:
+        """Initialize job and create batches"""
+        try:
+            # Load model
+            model_manager = ModelManager(InferenceAPIClient(), self.model_version.api_identifier)
+
+            if not model_manager.ensure_model_loaded():
+                # TODO: shouldn't we be getting an exact error out of the api client that we can store?
+                self.set_error("Failed to load model")
+                return
+
+            # Create batches
+            batch_processor = BatchProcessor(BatchConfig())
+            urls = self.collection.curated_urls.all()
+            batches = batch_processor.create_batches(urls)
+
+            # Create external jobs for batches
+            for batch in batches:
+                self.create_external_job(batch)
+
+            if not self.external_jobs.exists():
+                self.set_error("No batches created")
+                return
+
+            self.status = InferenceJobStatus.PENDING
+            self.save()
+
+        except Exception as e:
+            self.set_error(str(e))
+
+    def create_external_job(self, batch_data) -> "ExternalJob":
+        """Create and submit an external job for a batch"""
+        try:
+            api_client = InferenceAPIClient()
+
+            # Submit batch to API using model version identifier
+            job_id = api_client.submit_batch(self.model_version.api_identifier, batch_data)
+
+            if not job_id:
+                # TODO: can't we get an exact error out of the api client?
+                self.set_error("Failed to get job ID from API")
+                return None
+
+            # Create external job record
+            return ExternalJob.objects.create(
+                inference_job=self,
+                external_job_id=job_id,
+                url_ids=[item["url_id"] for item in batch_data],
+                status=ExternalJobStatus.QUEUED,
+            )
+
+        except Exception as e:
+            self.set_error(f"Failed to create external job: {str(e)}")
+            return None
+
+    def process_external_jobs(self) -> None:
+        """Process all pending external jobs"""
+        pending_jobs = self.external_jobs.filter(status__in=[ExternalJobStatus.QUEUED, ExternalJobStatus.PENDING])
+
+        for external_job in pending_jobs:
+            external_job.process()
+
+    def evaluate_status(self) -> None:
+        """Evaluate overall job status and handle completion"""
+
```

Comment:
we should separate this so that we have separate methods.
- checkers: checks and return maybe either bools or None vs a query set. 
- updaters: modify InferenceJob status.